compiler/next/lib/queries/Context.cpp

```diff
@@ -569,49 +569,66 @@ void Context::error(ErrorMessage error) {
   }
 }
 
-void Context::error(Location loc, const char* fmt, ...) {
+static void logErrorInContext(Context* context, Location loc,
+                              ErrorMessage::Kind kind,
+                              const char* fmt,
+                              va_list vl) {
   ID id;
   ErrorMessage err;
-  va_list vl;
-  va_start(vl, fmt);
-  err = ErrorMessage::vbuild(id, loc, fmt, vl);
-  va_end(vl);
-  Context::error(err);
+  err = ErrorMessage::vbuild(id, loc, kind, fmt, vl);
+  context->report(err);
 }
 
-void Context::error(ID id, const char* fmt, ...) {
-  Location loc = parsing::locateId(this, id);
+static void logErrorInContext(Context* context, ID id,
+                              ErrorMessage::Kind kind,
+                              const char* fmt,
+                              va_list vl) {
+  Location loc = parsing::locateId(context, id);
   ErrorMessage err;
-  va_list vl;
-  va_start(vl, fmt);
-  err = ErrorMessage::vbuild(id, loc, fmt, vl);
-  va_end(vl);
-  Context::error(err);
+  err = ErrorMessage::vbuild(id, loc, ErrorMessage::ERROR, fmt, vl);
+  context->report(err);
 }
 
-void Context::error(const uast::ASTNode* ast, const char* fmt, ...) {
-  Location loc = parsing::locateAst(this, ast);
+static void logErrorInContext(Context* context,
+                              const uast::ASTNode* ast,
+                              ErrorMessage::Kind kind,
+                              const char* fmt,
+                              va_list vl) {
+  Location loc = parsing::locateAst(context, ast);
   ErrorMessage err;
-  va_list vl;
-  va_start(vl, fmt);
-  err = ErrorMessage::vbuild(ast->id(), loc, fmt, vl);
-  va_end(vl);
-  Context::error(err);
+  err = ErrorMessage::vbuild(ast->id(), loc, ErrorMessage::ERROR, fmt, vl);
+  context->report(err);
+}
+
+#define CHPL_CONTEXT_LOG_ERROR_HELPER(context__, pin__, kind__, fmt__) \
+  do { \
```

Comment:
How about using just regular `{` `}` instead of the do/while?

Reply:
I'm not a really big fan of seeing macro invocations without a semicolon so I'll leave things as is unless we decide to flip the other way in the future.

Reply:
Works for me, not a big deal.